### PR TITLE
12632 - Broaden search for "solo" PlayerSide.

### PR DESCRIPTION
**Pull request by riverwanderer**
Attempt improved side switch filtering, for issue #12632.

### CODE REVIEW

**Comment by riverwanderer** [comment]
Tested successfully using C&C Ancients module for skipping names starting "Solo:" and "." as well as equals "Solitaire".

**Comment by riverwanderer** [comment]
Doco update pending.

**Comment by riverwanderer** [comment]
Component reference manaul [updated](file:///Applications/VASSAL-3.7.0-SNAPSHOT-8e34ce6-12632_moreSoloSides.app/Contents/Resources/doc/ReferenceManual/GameModule.html#Definition_of_Player_Sides).

Changed special prefix from dot to space.

Re-tested OK.

**Comment by riverwanderer** [comment]
Further change avoids potential conflict with ChessClock use by creating a dedicated IsRefereeSide() function, separate from IsSoloSide().

File: vassal-doc/src/main/readme-referencemanual/ReferenceManual/GameModule.adoc
Comment (BrentEaston) [inline vassal-doc/src/main/readme-referencemanual/ReferenceManual/GameModule.adoc:121]: Why a Global Translatable message?

Comment (riverwanderer) [reply vassal-doc/src/main/readme-referencemanual/ReferenceManual/GameModule.adoc:121]: It was my solution to the question of what happens if Translation(s) are added to the Definition of Sides component.
As the list of sides is translated, VassalSkipSideList will need to be translated also... or is there a better way ?

Comment (riverwanderer) [reply vassal-doc/src/main/readme-referencemanual/ReferenceManual/GameModule.adoc:121]: > Why a Global Translatable message?

Now that Ive learnt a bit more about side translation in PlayerRoster, I want to take another go at this.

Comment (riverwanderer) [reply vassal-doc/src/main/readme-referencemanual/ReferenceManual/GameModule.adoc:121]: To allow the PR to proceed, I will withdraw the extendable (VassalSkipSideList) element from this change and leave that to the wider design of side control features.

I'll also tackle the issue of side names that remain untranslated in a translated module.

File: vassal-app/src/main/java/VASSAL/build/module/PlayerRoster.java
Comment (Cattlesquat) [inline vassal-app/src/main/java/VASSAL/build/module/PlayerRoster.java:638]: Are these really needed? If a translated module hasn't localized the side names don't they come through in English anyway?

File: vassal-app/src/main/java/VASSAL/build/module/PlayerRoster.java
Comment (Cattlesquat) [inline vassal-app/src/main/java/VASSAL/build/module/PlayerRoster.java:635]: Why "startsWith" here where it used to only check equals? Are we expecting "Moderator Left" and "Moderator Right"? 

If so, why are we requiring a different format here from Solo (where it requires the colon e.g. "Solo:")? And for that matter why do we only eg allow "Solo:" and not "Solitaire:"

File: vassal-app/src/main/java/VASSAL/build/module/PlayerRoster.java
Comment (Cattlesquat) [inline vassal-app/src/main/java/VASSAL/build/module/PlayerRoster.java:651]: Again I'm fuzzy on the change to startsWith and also the need to compare to the raw strings

File: vassal-app/src/main/resources/VASSAL/i18n/VASSAL.properties
Comment (Cattlesquat) [inline vassal-app/src/main/resources/VASSAL/i18n/VASSAL.properties:1034]: This appears not to be used?

**Review by Cattlesquat — CHANGES_REQUESTED** [review]
See code notes -- I'm fuzzy on a few things

File: vassal-app/src/main/java/VASSAL/build/module/PlayerRoster.java
Comment (riverwanderer) [reply vassal-app/src/main/java/VASSAL/build/module/PlayerRoster.java:651]: The raw strings comparison arises from a concern that I have about this along the following lines...

1. Say a module has partial translation but not in the Definition of Sides component. This is not so outlandish, as translated PlayerSides can open up other complications if the module depends on PlayerSide, so I've found.
2. The translation resources for solitaire etc exist regardless.... therefore when Vassal isSoloSide() scans the sides array it will attempt to find the translated side name. This will fail if the solo side names don't match in translation, but the raw string will then catch it.

Have I got that right?

**Comment by riverwanderer** [comment]
This PR doesn't create any v3.7 conflicts, can it be re-classified for the 3.7.1 milestone?

File: vassal-app/src/main/java/VASSAL/build/module/PlayerRoster.java
Comment (BrentEaston) [reply vassal-app/src/main/java/VASSAL/build/module/PlayerRoster.java:638]: I think the actual problem is in the ChessClock which calling isSoloSide with the translated Sides, it should be checking untranslated sides, but displaying translated sides,

The concern is that ChessClocks now have translated sides recorded in them?, they should be untranslated sides.

Comment (Cattlesquat) [reply vassal-app/src/main/java/VASSAL/build/module/PlayerRoster.java:638]: @BrentEaston So you think this usage is okay? Also are you okay with the various changes to "startsWith"?

Comment (Cattlesquat) [reply vassal-app/src/main/java/VASSAL/build/module/PlayerRoster.java:638]: @BrentEaston And we're comparing both Resources.getString("Some String") and the literal "Some String" in here. Surely only one is right? Or is one way being left in because ChessClock is broken? (are other things broken too?)

**Comment by riverwanderer** [comment]
I am putting this PR on hold as #12732 & #12743 offer more generic and flexible solutions to the issue that this PR seeks to address.

**Pull request closed without merging**
